pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +405,62 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
+	errResp, _ := unwrapError(body)
+	if client.WaitOnJob { // check sync flag
+		unwrapMessage := &ErrorResponse{}
+		err := json.Unmarshal(body, unwrapMessage)
+		if err != nil {
+			return nil, err
+		}
+		ids, err := handleAsyncJob(unwrapMessage, client)
+
+		if err != nil {
+			return nil, fmt.Errorf("http response error: status (202), messages: %v", err.Error())
+		}
+		return ids, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+	}
+	return nil, fmt.Errorf("http response error: status (202), messages: %v", errResp)
```

Comment:
we arent extracting job id from error. job id is returned as second argument.

Reply:
we dont need to return error . returnig nil .

Reply:
It is fixed . We dont need to return error. this comment is appearing at multiplaces and have already replied .

Reply:
We provide the option for the user to set WaitOnJob to false.  The use model seems to be that the caller doesn't want to block while the job completes, and they would want to check for job completion asynchronously.  To do this, the caller will require the job ID.  I'd like to understand how the caller can access the job ID in this case.

volume_service_test.go always sets WaitOnJobs to true when creating the group service.  It would be great to see an example where WaitOnJobs is set to false.  Call an interface that returns a 202 status, get the job ID in return, and then check the job status using a separate call into the SDK.

Reply:
Sorry I missed this comment earlier. We don’t really need to support this use case since customer can always invoke GetJobs services to get all the services  he wants to monitor separately.

Reply:
How does the user know which job is the one that he wants to monitor?  I.e., the one that was created from this request.

Reply:
It contains object I’d as well

Reply:
How does the object ID in the job help me identify the job that was created from my POST request?

Reply:
Roger is right here.  We should provide an example of how a client would get the job_id off of the error response and then use it in GetJobs for subsequent monitoring.

Reply:
I should have been more explicit. If the user selects sync=false and the API response is HTTP Code 202 , in that case we return an error to the caller and the error message has job_id in it . The user has to extract the job_id and monitors it. 
Moreover, he can also invoke GetJob() service separately to monitor the respective job. the response contains job_id, name , operation type , object_id etc . it has enough information to filter out the required job.

Reply:
The error message comes back in string form, right?  It doesn't look so easy to extract the job ID.

How does one determine that the "error" really just means the response will be asynchronous?  It seems unintuitive, since it isn't really an error condition.

In general is there a way for the calling code to reliably determine the HTTP status code in an error case?  For example, how should a long-lived client application handle an auth token expiration?

Reply:
We don’t have concept of exception here with golang and in fact with other sdk it generates exception and user has to handle it. It is not intuitive but if user is interested in monitoring the job by himself he can extract the job ID from error string and it is not difficult

Reply:
I believe @rkumpf is drawing attention to the fact that the current error we report makes it difficult to determine you have an asynchronous job that you now have to monitor.  The Nimble REST API will return an error in this form:

```
// ErrorResponse is a serializer struct for representing a valid JSON API errors payload.
type ErrorResponse struct {
	Messages []*Message json:"messages"
}
// Message is an Error implementation as well as an implementation of the JSON API error object.
type Message struct {
	Code string json:"code,omitempty"
	Text string json:"text,omitempty"
}
```

And our error messages are reported in this form:

`fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), errResp)`

in order to fit within the Error interface that is provided by golang.  This is a valid concern and worthy of improvement.  There is a lot of discussion on this topic.  Here's one link worth looking into: https://middlemost.com/failure-is-your-domain/.  However, I feel our current implementation will suffice for now.  Can we file an issue to track this as an area of improvement in a future PR?  Thanks.

Reply:
sure thanks